### PR TITLE
refactor: decouple python from js and css

### PR DESCRIPTION
**Pull request by codedealer**
This is a large change that concerns `frontend` folder.
 In short `css_and_js.py` was refactored to be a facade for frontend files which are stored in their respective folders: `css` and `js`.

Currently the implementation is very simple. It bundles together css files based on ops and js is just one big file that exposes SD object to the global space. At a later date it can be expanded as needed while maintaining a simple interface with python scripts via `css_and_js.py`.

I tested it to my ability but the changes are vast so I might have missed a few things.

**IMPORTANT NOTES**
1. To load the js on load event I currently use a hack with invisible element. This is only temporary until the next gradio version is released. My PR has already been merged into master there: gradio-app/gradio#2108 
2. Calls to js functions depend on ids of the relevant components. Those should be made into dictionary or something like that instead of just string literals.
3. I had to remove the changes made in 7ddbd6533a03b3abd513462dacc676640b8499c0 in order to push. I will try to incorporate them later.

### CODE REVIEW

**Comment by hlky** [comment]
As this is such a large change I'll wait for you to resolve the conflicts, we'll get a few people to test then coordinate the sync to master because it's a big change of frontend again that will 💯% require a pull from main repo

**Comment by hlky** [comment]
Well nothing seems broken, tested txt2img and img2img 🤞

File: frontend/css_and_js.py [hunk 1]
```diff
@@ -1,67 +1,38 @@
...
+def css(opt):
+    styling = readTextFile("css", "styles.css")
+    if not opt.no_progressbar_hiding:
+        styling += readTextFile("css", "no_progress_bar.css")
+    return styling
 
-    input[type=number]:disabled { -moz-appearance: textfield;+ }
+def js(opt):
+    data = readTextFile("js", "index.js")
+    data = "(z) => {" + data + "; return z ?? [] }"
```
Comment (altryne) [inline frontend/css_and_js.py:19]: What's this part?

File: frontend/css_and_js.py [hunk 2]
```diff
@@ -1,67 +1,38 @@
...
-const getIndex = function(){
-    const selected = root.querySelector('#img2img_gallery_output .\\\\!ring-2');
-    return selected ? [...selected.parentNode.children].indexOf(selected) : 0;
-};"""
-js_part_clear_img2img="""
-root.querySelector('#img2img_editor .modify-upload button:last-child')?.click();
+def js_move_image(from_id, to_id):
+    return w(f"moveImageFromGallery('{from_id}', '{to_id}')")
 
+def js_copy_to_clipboard(from_id):
```
Comment (altryne) [inline frontend/css_and_js.py:31]: Python bindings for every JS function?

File: frontend/frontend.py [hunk 3]
```diff
@@ -396,4 +345,10 @@ def img2img_submit_params():
         If you would like to contribute to development or test bleeding edge builds, you can visit the <a href="https://github.com/hlky/stable-diffusion-webui" target="_blank">developement repository</a>.</p>
     </div>
     """)
+        # Hack: Detect the load event on the frontend
+        # Won't be needed in the next version of gradio
+        # See the relevant PR: https://github.com/gradio-app/gradio/pull/2108
```
Comment (altryne) [inline frontend/frontend.py:350]: 👏

**Pull request merged**